File: level2/삼각달팽이.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) {
    int top = 0, bottom = n-1, cur = 0;
    vector<vector<int>> v;
    for(int i = 1; i <= n; i++){
        vector<int> tmp;
        for(int j = 0; j < i; j++){
            tmp.push_back(0);
        }
        v.push_back(tmp);
    }
    
    int num = 0;
    while(top <= bottom){
        for(int i = top; i <= bottom; i++){
            num++;
            v[i][cur] = num;
        }
        for(int i = cur+1; i < bottom-cur; i++){
            num++;
            v[bottom][i] = num;
        }
        
        for(int i = bottom; i > top; i--){
            num++;
            v[i][i-cur] = num;
        }
        cur += 1;
        top += 2;
        bottom -= 1;
    }
    
    vector<int> answer;
    for(int i = 0; i < n; i++){
        for(int j = 0; j <= i; j++){
            answer.push_back(v[i][j]);
        }
    }
    return answer;
}

File: level2/삼각달팽이.py
def solution(n):
    answer = [[0]*i for i in range(1,n+1)]
        
    num = 0
    x_start, x_end, y_start, y_end = 0, n-1, 0, n-1
    while x_start <= x_end:
        print(x_start, x_end, y_start, y_end)
        for i in range(x_start,x_end+1):
            num += 1
            answer[i][y_start] = num
            
        for i in range(y_start+1,y_end):
            num += 1
            answer[x_end][i] = num
        
        tmp = y_end
        for i in range(x_end,x_start,-1):
            num += 1
            answer[i][tmp] = num
            tmp -= 1
        
        x_start += 2
        x_end -= 1
        y_start += 1
        y_end -= 2
        
    return sum(answer, [])
